clippy_lints/src/almost_complete_letter_range.rs

```diff
@@ -79,6 +79,7 @@ fn check_range(cx: &EarlyContext<'_>, span: Span, start: &Expr, end: &Expr, sugg
             (LitKind::Byte(b'a') | LitKind::Char('a'), LitKind::Byte(b'z') | LitKind::Char('z'))
             | (LitKind::Byte(b'A') | LitKind::Char('A'), LitKind::Byte(b'Z') | LitKind::Char('Z'))
         )
+        && !span.from_expansion()
```

Comment:
It would be better to detect only macros from foreign crates.
```suggestion
        && !in_external_macro(cx.sess(), span)
```


This commit would be helpful to implement tests for external macros: df26c3f551fec5854e3acd140782b8f8c98e987b